immuneML/encodings/pdbencoding/PDBEncoder.py

```diff
@@ -0,0 +1,250 @@
+
+
+                    saveList = list.copy()
+                    collection.append(saveList)
+                    counter = counter + 1
+                    list.clear()
+
+            lightID = -1
+            heavyID = -1
+            antigenID = -1
+
+            for i in range(0,len(collection)):
+                chainID = collection[i][0].id
+
+                if 'L' in chainID or 'A' in chainID:
+                    lightID = i
+
+                elif 'H' in chainID or 'B' in chainID:
+                    heavyID = i
+
+                elif 'P' in chainID or 'C' in chainID:
+                    antigenID = i
+
+                collection[i].pop(0)
+
+
+            if lightID >= 0 and heavyID >= 0 and antigenID == -1:
+                if lightID + heavyID == 1:
+                    antigenID = 2
+
+                elif lightID + heavyID == 3:
+                    antigenID = 0
+
+            elif lightID == -1  or heavyID == -1 or antigenID == -1:
+                lightID = 0
+                heavyID = 1
+                antigenID = 2
+
+
+            lightChainDistanceToAntigen = np.zeros(shape=(len(collection[antigenID]), len(collection[lightID])))
+
+            caluculateDistance(collection, lightID, antigenID, lightChainDistanceToAntigen)
+
+            heavyChainDistanceToAntigen = np.zeros(shape=(len(collection[antigenID]), len(collection[heavyID])))
+            caluculateDistance(collection, heavyID, antigenID, heavyChainDistanceToAntigen)
+
+            paddedArrays = paddingOfArray(lightChainDistanceToAntigen,heavyChainDistanceToAntigen)
+
+            structures.append([paddedArrays[0],paddedArrays[1]])
+            CDRcounter = CDRcounter +1
+
+
+        longestAntigen = 0
+        longestChain = 0
+
+        for x in structures:
+            if len(x[0]) > longestAntigen:
+                longestAntigen = len(x[0])
+            if len(x[0][0]) > longestChain:
+                longestChain = len(x[0][0])
+
+        paddedStructures = []
+        for x in structures:
+            paddedStructures.append(expandAntibodyChainToMaxLength(x, longestAntigen, longestChain))
+
+
+
+        numpArray = np.array(paddedStructures)
+
+        encoded_dataset = PDBDataset(self, dataset.pdbFilePaths,dataset.labels,dataset.metadata_file,EncodedData(numpArray,None,dataset.get_example_ids()))
+
+        #print("Shape of encoded_dataset.encoded_data.examples: " , numpy.shape(encoded_dataset.encoded_data.examples))
+        return encoded_dataset
+
+
+def caluculateDistance(collection, chain, antigen, list):
+    for x in range(len(collection[antigen])):
+        for i in range(len(collection[chain])):
+            a = np.array((collection[chain][i][0], collection[chain][i][1], collection[chain][i][2]))
+            b = np.array((collection[antigen][x][0], collection[antigen][x][1], collection[antigen][x][2]))
+
+            rounded = round(np.linalg.norm(a - b), 2)
+            list[x][i] = rounded
+
+
+"""
```

Comment:
triple quotes are reserved for docstrings, and docstrings should be placed immediately in a function/method/class